shellbase-core/src/main/scala/com/sumologic/shellbase/ShellCommand.scala

```diff
@@ -100,6 +106,15 @@ abstract class ShellCommand(val name: String,
         false
       }
     } finally {
+      import scala.concurrent.ExecutionContext.Implicits.global
```

Comment:
Let's not force the global execution context on everyone.  We should add an implicit argument to the class's constructor, and default to this for backwards compatibility reasons.

Context:  Internally, we've seen ourselves fill up the global execution pool (not necessarily in shells though) and we should make it easy to avoid

Reply:
this would change the constructor though. Do we want that?